Sources/NIOPosix/Bootstrap.swift

```diff
@@ -1070,6 +1082,20 @@ public final class NIOPipeBootstrap {
     }
 
     private func validateFileDescriptorIsNotAFile(_ descriptor: CInt) throws {
+#if os(Windows)
+        // NOTE: this is a *non-owning* handle, do *NOT* call `CloseHandle`
+        let hFile: HANDLE = HANDLE(bitPattern: _get_osfhandle(descriptor))!
+        if hFile == INVALID_HANDLE_VALUE {
+            throw IOError(errnoCode: EBADF, reason: "_get_osfhandle")
+        }
+
+        switch GetFileType(hFile) {
+        case DWORD(FILE_TYPE_PIPE):
```

Comment:
This is not quite the semantic of the non-Windows code, which just checks whether things are not a regular file, a directory, a symlink, or a block device. Sockets, character devices, and pipes all pass this check. Is it possible to replicate this logic?

Reply:
The sockets and file descriptors are completely separate domains, so its implicitly not a socket.

I suppose that we could add the check for the character devices and block devices, though, I'm worried that it is less than useful as pipes are completely different from the other file types (not to mention different from Unix-like platforms).

Reply:
I'm ok leaving those checks out, but I think we need a code comment explaining _why_ the Windows code is written the way it is. Otherwise if it misbehaves in future I'll have no idea why the two branches do two (apparently) utterly different things.